x-pack/plugins/fleet/server/routes/message_signing_service/handlers.ts

```diff
@@ -18,22 +18,31 @@ export const rotateKeyPairHandler: FleetRequestHandler<
   TypeOf<typeof RotateKeyPairSchema.query>,
   undefined
 > = async (_, __, response) => {
+  const messageSigningService = appContextService.getMessageSigningService();
+  if (!messageSigningService) {
+    return response.customError({
+      statusCode: 500,
+      body: {
+        message: 'Failed to rotate key pair. Message signing service is unavailable!',
+      },
+    });
+  }
+
   try {
-    const rotateKeyPairResponse = await appContextService
-      .getMessageSigningService()
-      ?.rotateKeyPair();
+    const rotateKeyResponse = await messageSigningService.rotateKeyPair();
 
-    if (!rotateKeyPairResponse) {
-      return response.customError({
-        statusCode: 500,
+    if (!rotateKeyResponse.error && rotateKeyResponse.success) {
+      return response.ok({
         body: {
-          message: 'Failed to rotate key pair!',
+          message: 'Key pair rotated successfully.',
         },
       });
     }
-    return response.ok({
+
+    return response.customError({
```

Comment:
Suggestion: `log.error()` the entire `rotateKeyResponse` for debug purposes. Hoping that has more details around the error encountered

Reply:
Also - its a bit strange (to me) to have a call to a service return an `error` and not actually `throw`.